### ISSUE TITLE
[Feature] Support .m3u files for multi-disk games

### ISSUE THREAD

**Issue by SatorSquare** [rommapp/romm#1298]
**Is your feature request related to a problem? Please describe.**
Currently, RomM does not support using .m3u files to define multipart games. RomM will only recognize multipart games if they are all contained within the same subfolder. It would be useful to be able to define multipart games via .m3u files regardless of file-structure.

**Describe the solution you'd like**
RomM should recognize multipart games defined via .m3u files. This would be useful for the following platforms:

- [ ] PC-FX
- [ ] Playstation
- [ ] Sega CD
- [ ] Sega Saturn

**Comment by gantoine** [comment]
We're not going to be supporting this any time soon, as romm generates its own m3u files for multi-rom games. If we do implement this, it'll be in sync with other work around PS1 support.

**Comment by SatorSquare** [comment]
@gantoine are there any plans to support other methods of identifying multi-rom games aside from relying on folder structure? For example, allowing them to be tagged as such in the UI, or detecting them from file name like other rom managers?

**Comment by gantoine** [comment]
> detecting them from file name like other rom managers?

what would the structure look like? if you can give an example that would be helpful. oh and other rom managers that support flat file structure.

**Comment by SatorSquare** [comment]
Absolutely! The one I'm primarily referring to is LaunchBox. When your roms are scanned, all roms with the same name (without tags) are considered the same game. Then, the games' files are sub-grouped by their region/version tags similarly to how RomM does now. Finally, if "(Disk #)" tags are detected in the filename the game is labeled as a multi-part game and its files are grouped further by those disk tags. When a game is run, a temporary .m3u file is created and automatically passed to the appropriate emulator for the platform. This is _extremely_ useful for supporting larger collections automatically.

There is a bit more nuance since LaunchBox will honor, and properly associate pre-existing .m3u's that you have already created. It also handles bin/cue files and systems where roms are more than one file. Additionally, there are per-platform settings that allow control over this, etc...

**Comment by gantoine** [comment]
Ok I'm going to re-open this and backlog it, but under the scope of grouping files same-name-same-game, and supporting `(Disk #)`.

> When your roms are scanned, all roms with the same name (without tags) are considered the same game.

We'd probably just keep the current behavior for since it works well as-is, for games with different tags.

> If "(Disk #)" tags are detected in the filename the game is labeled as a multi-part game and its files are grouped further by those disk tags

What we'd likely do here is move those into a subfolder on import, to match the existing structure. There's no system out there that requires files be in a flat structure when multi-files games are supported, so it shouldn't break anything.

> When a game is run, a temporary .m3u file is created and automatically passed to the appropriate emulator for the platform

The app already does this for non-compressed multi-file games (ie not a single ZIP, but a folder with multiple ISO/etc).

> properly associate pre-existing .m3u's that you have already created

If an .m3u file already exists in the multi-part game folder, a temp one won't be created.

I think it's important to note that RomM and Launchbox aren't building the same solution, and the fact that LB handles bin/cue files, reads the contents on .m3u files, allows custom configs, and other things it because it also acts as a frontend for your emulators. That being said there is work we can and will do to improve interop between both systems.

**Comment by gantoine** [comment]
Well with 3.8 now released and the work we're done of multi-disk file structure support I think most of these issues are now addressed/moot. As long as you separate each game into it' own subfolder the behaviour you're expecting should be the default.
